shared/public/system.h

```diff
@@ -121,6 +122,47 @@ namespace ink
 		template<typename T>
 		struct always_false { static constexpr bool value = false; };
 	}
+
+#ifdef INK_ENABLE_STL
```

Comment:
I think this is smart. Use the familiar STL optional when we have it enabled and roll our own otherwise. At first I was thinking "why not just always use our own if it's here" but if the user is using STL they'd probably prefer the familiar class instead of one masquerading with the same name.

Reply:
Im one of this people, who find it anyoing when a lib has an type which also exist in stl and function slidly differnt ^^